hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionWrapperImpl.java

```diff
@@ -357,6 +364,32 @@ public void run() {
     }
   }
 
+  @Override
+  public String getTableDescriptorHash() {
+    return tableDescriptorHash;
+  }
+
+  private String computeTableDescriptorHash() {
+    try {
+      TableDescriptor tableDesc = this.region.getTableDescriptor();
```

Comment:
Wrap the computation in an LRU cache to avoid recomputing this 100's of times for all regions of the same table open on one host?